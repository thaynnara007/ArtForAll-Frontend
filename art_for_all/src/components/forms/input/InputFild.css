.space{
    margin-bottom: 0.8em;
    margin-top: 0.8em;
}

