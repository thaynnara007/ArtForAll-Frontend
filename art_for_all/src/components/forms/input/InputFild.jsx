import React from 'react';

class InputFild extends React.Component{

    render(){
        return(
            <div className="input-group mb-3">
                <input type="text" className="form-control" placeholder={this.props.placeHolder} aria-label="Recipient's username" aria-describedby="basic-addon2" onChange={(event) => this.props.handleChange(event, this.props.data)}></input>
            </div>
        )
    }
}

export default InputFild;