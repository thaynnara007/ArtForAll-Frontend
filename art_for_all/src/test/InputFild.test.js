import React from 'react';
import Jest from 'jest';
import Enzyme from './Enzyme';
import { shallow} from 'enzyme';
import InputFild from '../components/forms/input/InputFild';

describe("INPUT FILD - Testing with snapshots", () =>{

    it('TEST01:render correctly input fild component with type=text', () => {  
        const InputFildComponent = shallow(<InputFild />);
        expect(InputFildComponent).toMatchSnapshot();
    });

    it('TEST02:render correctly input fild component with type=password', () => {  
        const InputFildComponent = shallow(<InputFild type="password"/>);
        expect(InputFildComponent).toMatchSnapshot();
    });

    it('TEST03:render correctly input fild component with placeHolder=newPlaceHolder and type=password', () => {  
        const InputFildComponent = shallow(<InputFild type="password" placeHolder="newPlaceHolder"/>);
        expect(InputFildComponent).toMatchSnapshot();
    });

    it('TEST03:render correctly input fild component with placeHolder=newPlaceHolder and type=text', () => {  
        const InputFildComponent = shallow(<InputFild type="text" placeHolder="newPlaceHolder"/>);
        expect(InputFildComponent).toMatchSnapshot();
    });
})