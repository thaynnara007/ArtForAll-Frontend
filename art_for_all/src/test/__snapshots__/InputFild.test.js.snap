// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INPUT FILD - Testing with snapshots TEST01:render correctly input fild component with type=text 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-describedby="basic-addon2"
    aria-label="Recipient's username"
    className="form-control"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`INPUT FILD - Testing with snapshots TEST02:render correctly input fild component with type=password 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-describedby="basic-addon2"
    aria-label="Recipient's username"
    className="form-control"
    onChange={[Function]}
    type="password"
  />
</div>
`;

exports[`INPUT FILD - Testing with snapshots TEST03:render correctly input fild component with placeHolder=newPlaceHolder and type=password 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-describedby="basic-addon2"
    aria-label="Recipient's username"
    className="form-control"
    onChange={[Function]}
    placeholder="newPlaceHolder"
    type="password"
  />
</div>
`;

exports[`INPUT FILD - Testing with snapshots TEST03:render correctly input fild component with placeHolder=newPlaceHolder and type=text 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-describedby="basic-addon2"
    aria-label="Recipient's username"
    className="form-control"
    onChange={[Function]}
    placeholder="newPlaceHolder"
    type="text"
  />
</div>
`;
